Update `Article styling criteria/Formatting` with infobox usage (#8063)
ppy/osu-wiki: wiki/Article_styling_criteria/Formatting/en.md

Images should use the inline linking style. Reference link definitions must be placed at the bottom of the article.

> All inline images on the page are combined into a single gallery, which can be navigated using arrow icons on both sides of the screen, keyboard shortcuts, or screen swipes on mobile devices<!-- https://github.com/ppy/osu-web/pull/8126 -->.

Reviewer: ```suggestion
All inline images on the page (that have nothing else on the same line) are combined into a single gallery, which can be navigated using arrow icons on both sides of the screen, keyboard shortcuts, or screen swipes on mobile devices<!-- https://github.com/ppy/osu-web/pull/8126 -->.
```

> All inline images on the page (that have nothing else on the same line) are combined into a single gallery, which can be navigated using arrow icons on both sides of the screen, keyboard shortcuts, or screen swipes on mobile devices<!-- https://github.com/ppy/osu-web/pull/8126 -->.

Reviewer 2: ```suggestion
All block images on the page (that have nothing else on the same line) are combined into a single gallery, which can be navigated using arrow icons on both sides of the screen, keyboard shortcuts, or screen swipes on mobile devices<!-- https://github.com/ppy/osu-web/pull/8126 -->.
```

should be more semantically correct this way i think

If an image is the sole content of a paragraph, it displays as a centred block. Otherwise, it flows with the surrounding inline text.

> Block images with title text display the title text as a caption below the image. The caption will not be rendered if there's an [HTML comment](#comments) on the same line.

Reviewer: ```suggestion
Block images with title text display the title text as a caption below the image. Avoid adding [HTML comment](#comments) or any other text on the same line as the image, as this will cause the caption to not be rendered.
```

> Block images with title text display the title text as a caption below the image. Avoid adding [HTML comment](#comments) or any other text on the same line as the image, as this will cause the caption to not be rendered.

Reviewer 2: ```suggestion
Block images with title text display the title text as a caption below the image. Avoid adding [HTML comment](#comments) or any other text on the same line as the image, as this will cause the caption not to be rendered.
```

| Maria | No Contest | Mocha |  |
```

## Infoboxes

An infobox is a fixed-width block which is aligned to the right side of the article. It may contain a relevant image, which explains the surrounding text, or a block of navigation that links to other articles from the same category.

Example use, rendered on the right:

<!-- The real infobox is added for illustrative purposes, with Markdown syntax duplicated below for clarity. -->

::: Infobox
![](/wiki/shared/mods/SD.png "Sudden Death mod icon")
:::

```markdown
::: Infobox
![](/wiki/shared/mods/SD.png "Sudden Death mod icon")
:::
```

> The title text serves as the image caption. If an image inside the infobox is wider than 300px<!-- https://github.com/ppy/osu-web/blob/bca69155332cb0c020ba225f2f9de579ad99284c/resources/assets/less/bem/osu-md.less#L226 -->, its thumbnail will be displayed instead, with full version available on click. Images from all infoboxes on the page are combined into a single gallery, which can be navigated using arrow icons on both sides of the screen<!-- https://github.com/ppy/osu-web/pull/8126 -->.

Reviewer: >The title text serves as the image caption.

that's a general thing - a single image on its own line will have its title text displayed as a caption. infoboxes are like tiny little markdown sections

> Images from all infoboxes on the page are combined into a single gallery, which can be navigated using arrow icons on both sides of the screen

*any* image that is on its own are viewable in the gallery, again this doesn't only apply to infoboxes

(+arrow keys, swiping, etc. also work for navigation)

Author: didn't know that. guess the whole paragraph needs to be removed, or partially moved into the images section

### Pitfalls

> - Do not use infoboxes in short sections, since the next section's title will only appear after the last used infobox ends.

Reviewer: ```suggestion
- Do not use infoboxes in short sections. The next section's heading appears below any infoboxes, leaving a large gap between the text and the next heading.
```

> - Image titles followed by [HTML comments](#comments) are hidden.

Reviewer: more like html comments causes the image to not be interpreted as being the only thing on a line

Author: not sure how to phrase that better, except for "images followed by ..."

Reviewer: also should be put in the images section instead

> - If several images are not on the same line, including them in an infobox yields poor results.

Reviewer: wdym by this?

Author: it looked like shit last time I checked

Reviewer: this?

![image](https://user-images.githubusercontent.com/36758269/192868873-15bb9fc5-bf48-466f-a6db-1cc76a22eea9.png)

Author: yes, iirc it was the reason we had to put each screenshot into a separate infobox: https://github.com/ppy/osu-wiki/blob/master/wiki/Client/Interface/Multiplayer/en.md?plain=1#L132-L138
